formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt

```diff
@@ -280,6 +282,59 @@ class FormulaRuntimeTest(val runtime: TestableRuntime, val name: String) {
             .output { assertThat(state).isEqualTo(1) }
     }
 
+    @Test
+    fun `child action triggers parent event on start`() {
+        val increments = listOf(1, 1, 1, 1, 1, 1)
+        runtime.test(
+            formula = ChildActionFiresParentEventOnStart.formula(runChildOnStart = true, increments),
+            input = Unit,
+        )
+            .output { assertThat(value).isEqualTo(6) }
+            .apply {
+                // Efficiently we emit only emit 2 values
+                assertThat(values()).hasSize(1)
+            }
+
+        runtime.test(
+            formula = ChildActionFiresParentEventOnStart.formula(runChildOnStart = false, increments),
+            input = Unit,
+        )
+            .output { assertThat(value).isEqualTo(0) }
+            .output { showChild(true) }
+            .output { assertThat(value).isEqualTo(6) }
+            .apply {
+                // Efficiently we emit only emit 2 values
+                assertThat(values()).hasSize(2)
+            }
+    }
+
+    @Test
+    fun `on action start child triggers action in a parallel child`() {
+        val events = listOf(Unit, Unit, Unit, Unit)
```

Comment:
you could do `List(4) { Unit }` here and pull 4 out into a val that they can all reference

Reply:
Will update in follow-up